frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/GitMacheteRepositoryDiscoverer.java

```diff
@@ -0,0 +1,86 @@
+package com.virtuslab.gitmachete.frontend.ui.impl.table;
+
+import static com.intellij.openapi.application.ModalityState.NON_MODAL;
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString;
+
+import java.nio.file.Path;
+import java.util.function.Consumer;
+
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vcs.VcsNotifier;
+import com.intellij.ui.GuiUtils;
+import io.vavr.control.Try;
+import lombok.AllArgsConstructor;
+import lombok.CustomLog;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+
+import com.virtuslab.binding.RuntimeBinding;
+import com.virtuslab.branchlayout.api.BranchLayoutException;
+import com.virtuslab.gitmachete.backend.api.IGitMacheteRepositoryCache;
+import com.virtuslab.gitmachete.backend.api.IGitMacheteRepositorySnapshot;
+import com.virtuslab.gitmachete.frontend.ui.api.gitrepositoryselection.IGitRepositorySelectionProvider;
+import com.virtuslab.gitmachete.frontend.ui.providerservice.BranchLayoutWriterProvider;
+import com.virtuslab.gitmachete.frontend.vfsutils.GitVfsUtils;
+
+@AllArgsConstructor
+@CustomLog
+public class GitMacheteRepositoryDiscoverer {
+
+  private final Project project;
+  private final IGitRepositorySelectionProvider gitRepositorySelectionProvider;
+  private final Consumer<Path> onFailurePathConsumer;
+  private final Consumer<IGitMacheteRepositorySnapshot> onSuccessRepositoryConsumer;
+
+  @UIEffect
+  public void doDiscover(Path macheteFilePath) {
```

Comment:
Hmm I would call a method that has an async component (`queue()`) using the word `(en)queue...` (or #620 for the state-of-the-art-ish solution)

Reply:
So `do...` is discouraged coz it sounds like sth synchronous :/